ocaml/ocamltest/tsl_semantics.ml

```diff
@@ -92,6 +92,19 @@ type test_tree =
     string located list *
     (test_tree list)
 
+let tests_do_something (tests : Tests.t) =
+  List.exists Actions.does_something tests.test_actions
```

Comment:
(It is probably fine as it stands, but
shouldn't the list end with an action
that does something?)

Reply:
Hmm.  Maybe it should be: if the test has children (which iiuc means another line that follows, after a semicolon) then `test_actions` doesn't need to do something, so long as all of the children do.  However if the test doesn't have children then I think you are right, and we need to check the list.  Does that sound plausible?

Reply:
I think you are right, each possible string
should end with an action doing something,
but contrary to what I implied a string is
the sequence of actions from the root
to a leaf - concatenating the lists of actions
along the path.

Reply:
Should be fixed